app/Http/Controllers/WikiEntityImportController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\WikiEntityImportStatus;
+use Illuminate\Http\Request;
+use App\Wiki;
+use App\WikiEntityImport;
+use App\Jobs\WikiEntityImportDummyJob;
+use Carbon\Carbon;
+
+class WikiEntityImportController extends Controller
+{
+    public function get(Request $request): \Illuminate\Http\JsonResponse
+    {
+        $wiki = Wiki::find($request->input('wiki'));
+        if (!$wiki) {
+            abort(404, 'No such wiki');
+        }
+
+        $imports = $wiki->wikiEntityImports()->get();
+        return response()->json(['data' => $imports]);
+    }
+
+    public function create(Request $request): \Illuminate\Http\JsonResponse
+    {
+        $wiki = Wiki::find($request->input('wiki'));
+        if (!$wiki) {
+            abort(404, 'No such wiki');
+        }
+
+        $imports = $wiki->wikiEntityImports()->get();
+        foreach ($imports as $import) {
+            if ($import->status === WikiEntityImportStatus::Success) {
+                return response()
+                    ->json(['error' => 'Wiki "'.$wiki->domain.'" already has performed a successful entity import'])
```

Comment:
I'm not sure I understand this logic (or rather the intended function) right: is the idea that per wiki only one import is gonna be possible?

Reply:
Yes. This is what the ticket says

> the button to copy the entities can only be triggered once per wiki. after having copied them over, the button is disabled and a message appears informing the user that copying can only happen once

I'm not sure why that is though.

Reply:
Aha interesting! Okay then at least I didn't misunderstand it